dart/test/event_processor/deduplication_event_processor_test.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:sentry/sentry.dart';
+import 'package:sentry/src/event_processor/deduplication_event_processor.dart';
+import 'package:test/test.dart';
+
+import '../mocks.dart';
+
+void main() {
+  group('$DeduplicationEventProcessor', () {
+    test('deduplicates', () {
```

Comment:
I'm pretty sure I'm missing some test cases.
Which events need deduplication and which don't?
As far as I can tell, Events without an exception aren't deduplicated and only events with the same exception are deduplicated.

Reply:
Also maybe a test case for the max length of the map.

Reply:
the main use case that should be deduplicated is like:

```
  private void test() {
    try {
      final int test = 1/0;
    } catch (Exception e) {
      // `e` is captured
      Sentry.captureException(e);
      throw e;
    }
  }
  
  private void myTest() throws Exception {
    // `test` throws and UncaughtExceptionHandlerIntegration gets called capturing `e` again
    // as `e` was already captured in the method `test`, the 2nd capture should be deduplicated
    test();
  }
```